Updates and all associated links in the glossary in sdk docs v5.1.4 - Closes #1189 (#1228)
LiskHQ/lisk-docs: modules/ROOT/pages/glossary.adoc

:url_token_unlock: protocol:transactions.adoc#unlock
:url_data_structure: advanced-explanations/schemas.adoc#data-types
:url_hash_onion: guides/node-management/forging.adoc#add-forging-data-to-config
:url_blockchain_app: introduction/blockchain-applications.adoc

> :url_mainnet: introduction/blockchain-applications.adoc

Reviewer: Lisk Mainnet is referring to the Lisk Core mainnet, not general blockchain applications

```suggestion
:url_mainnet: master@lisk-core::index.adoc#mainnet
```

Author: Ah yes...well spotted Mona! 👍

Addresses are unique identifiers for accounts in the Lisk blockchain.
They are the 20-byte hash value of the public key that controls the account initially.

> In all front-end products, addresses are displayed in a human-readable representation called <<Lisk32 representation>>.

Reviewer: this anchor link doesnt look right

Author: As discussed, it is now updated 👍

The Advanced Encryption Standard (AES), has a key length of 256 bits and is a block cipher adopted as an encryption standard by the U.S. government for military and government use. CBC (Cipher Blocker Chaining), is an advanced form of block cipher encryption. A detailed description of how AES functions can be found {url_aes}[here^].

== B

> === BCH Codes - Bose, Chaudhuri and Hocquenghem codes

Reviewer: BIP should be removed as well.

Blockchain application description should be improved, as it is very short, also, we have now a page dedicated to explain blockchain applications, the glossary should link to it.

Author: We agreed in the Glossary & Ambigous terminologies STM that we would keep BIP (See excel sheet in Trello card).
I have expanded the description of 'Blockchain Application' and also now added a link to the Blockcahin App page 👍

== I

== J

== K

== L

=== Lisk Framework

Lisk Framework is an application framework responsible for establishing and maintaining the interactions between the modules of a blockchain application compatible with the Lisk protocol. It provides an application where off-chain and on-chain logic can be extended with user defined plugins and modules.

=== LIP - Lisk Improvement Proposal

A LIP is a document usually structured by the research team defining technical changes in the Lisk protocol.
In addition, it can also cover an implementation or a non technical process surrounding Lisk.
Furthermore, a LIP also describes the requirements, rationale and motivation for the required changes.

=== Lisk SDK

The Lisk SDK is an open-source software development kit which enables developers to easily build scalable Blockchain applications using JavaScript.

=== Lisk32 representation

> The Lisk32 representation of an <<Account Address>> starts with the letters "lsk" followed by a custom Base32 encoding of the 20 bytes of the address appended with a BCH checksum.

Reviewer: wrong link format

Author: Corrected 👍

This is required because the public key is required to convert from the old address system to the new address system.
Non-initialized accounts don’t have a public key associated with them yet.

== M

=== Mainnet

> The official main network of Lisk.

Reviewer: Description could be improved, as it very short. Also, we have a page in the docs explaining the different networks, the glossary should link to it.

Author: Updated the mainnet description and added a link 👍

== R

=== Reducer

> A reducer is classed as a function which takes two arguments, namely an action and the previous state of the application, and returns the new state.

Reviewer: I don't think this description is correct

Author: I have amended it, please feel free to add/amend/update if you feel there is something else we could add here..?

== S

=== Standby Delegate

A standby delegate, as each delegate is not an <<active-delegate>>.
Rank 102-∞.

=== State Store

> The state store is a data structure that holds a temporary state while processing a block.

Reviewer: Description should be improved, as it is very short. Also, we are explaining the state store in the docs, the glossary can link to it.

Author: Done 👍

== T

=== Testnet

> The official testing network of Lisk.

Reviewer: Description is veery short, should be improved

Author: Have now updated the Testnet description, please feel free to add/amend if necessary 👍

== R

=== Reducer

> A reducer is classed as a function that determines changes to the state of an application. It can use the previous state and action to execute another state.

Reviewer: ```suggestion
Reducers are functions which can be invoked via Remote-Procedure-Calls (RPC) by other modules.
Reducers are able to perform state changes on the blockchain.
Modules and module assets can invoke reducers through the `reducerHandler`.
See xref:introduction.adoc#reducers[Modules > Reducers] for more information.
```

Author: Excellent! 👍

== S

=== Standby Delegate

A standby delegate, as each delegate is not an <<active-delegate>>.
Rank 102-∞.

=== State Store

> The xref:{url_state_store}[state store] is a data structure that holds a temporary state while processing a block. It is used to retrieve data from the blockchain, and exposes an interface to create, restore, and finalize snapshots. It is based on the following 3 states: Accounts, Chain and Consensus.

Reviewer: That the state store exposes `an interface to create, restore, and finalize snapshots` is new to me.
Also, the third state `consensus` seems new to me, and is also not reflected in the stateStore interface:
https://lisk.com/documentation/lisk-sdk/introduction/modules.html#the-state-store

Please double check if the explanation is correct.

Author: Thanks, I have reworded it accordingly 👍

> An interaction with the Lisk blockchain which permanently writes data to it.

Reviewer: Description here can be improved as well, e.g. describing which properties a transaction object has in general

Author: I have expanded on the description, please free free to add/amend if necessary 👍

Reviewer: I added my suggestion below!

=== Transaction

An interaction with the Lisk blockchain which permanently writes data to it.

> A transaction can have multiple purposes, e.g. a balance transfer or delegate registration, and posseses a common set of properties. Each transaction induces a specific operation on the accounts and is defined by a unique schema and transaction logic.

Reviewer: ```suggestion
A transaction can have multiple purposes, e.g. a balance transfer or delegate registration.
Each transaction contains a module ID and an asset ID, defining the transaction type.
All properties of a transaction object are described on the xref:protocol:transactions.adoc#transaction-properties[Lisk protocol > Transactions] page. The input data for the different <<module-asset, module assets>> is stored inside the <<transaction-asset>> of a transaction.
```
